src/sentry/profiles/task.py

```diff
@@ -227,17 +235,25 @@ def _normalize(profile: Profile, organization: Organization) -> None:
             profile["device_classification"] = classification
 
 
-def _prepare_frames_from_profile(profile: Profile) -> Tuple[List[Any], List[Any]]:
+def _prepare_frames_from_profile(profile: Profile) -> Tuple[List[Any], List[Any], List[int]]:
     with sentry_sdk.start_span(op="task.profiling.symbolicate.prepare_frames"):
         modules = profile["debug_meta"]["images"]
+        frames: List[Any] = []
+        frames_sent: List[int] = []
 
         # NOTE: the usage of `adjust_instruction_addr` assumes that all
         # the profilers on all the platforms are walking stacks right from a
         # suspended threads cpu context
 
         # in the sample format, we have a frames key containing all the frames
         if "version" in profile:
-            frames = profile["profile"]["frames"]
+            if profile["platform"] in ["javascript", "node"]:
```

Comment:
Maybe add a new frozenset for the javascript/node as it's reused more than once here.